airflow/api_fastapi/core_api/security.py

```diff
@@ -80,9 +80,11 @@ async def get_user_with_exception_handling(request: Request) -> BaseUser | None:
 
 def requires_access_dag(method: ResourceMethod, access_entity: DagAccessEntity | None = None) -> Callable:
     def inner(
+        request: Request,
         user: Annotated[BaseUser, Depends(get_user)],
-        dag_id: str | None = None,
     ) -> None:
+        dag_id = request.path_params.get("dag_id")
```

Comment:
Just realized my comment was missplaced, it should be here the:
```
or request.query_params.get("dag_id")
```

Sorry about that.

Reply:
updated. thanks!